src/casts/interpret-casts.rkt

```diff
@@ -28,7 +29,21 @@
     (define ic-expr! : (C0-Expr -> CoC3-Expr)
       (case (cast-representation)
         [(|Type-Based Casts|) (interpret-casts/type-based-casts)]
-        [(Coercions) (interpret-casts/coercions)]
+        [(Coercions)
+         (cond
+          [(enable-crcps?)
+           ;; avoid removing id coercions around injection/projection:
+           (optimize-first-order-coercions? #f)  ; should be parameterize?
```

Comment:
I rewrote as in d94e419.

I have to explain the background why we turned off `optimize-first-order-coercions?`.

In short, the following program (which requires user input e.g., 100) fails if `optimize-first-order-coercions?` is turned on.

```racket
(letrec ([even
            (lambda ([n : Int]) : Dyn
               (if (= 0 n)
                  #f
                  (odd (- n 1))))]
         [odd
            (lambda ([n : Int]) : Bool
               (if (= 0 n)
                  #t
                  (even (- n 1))))])
   (time (odd (read-int))))
```

In more detail: Coercions are inserted like: `(odd (n-1))<id;Bool!>`, `(even (n-1))<Bool?;id>`. With `optimize-first-order-coercions?`, they are optimized into `(odd (n-1))<Bool!>` and `(even (n-1))<Bool?>`. But, it seems that the composition `Bool! ;; Bool?` fails. So, I toggled off `optimize-first-ordercoercions?` so that the composition `(id;Bool!) ;;  (Bool?;id)` will happen.

That said, we also think that modifying the coercion composition function so that `Bool! ;; Bool? = id` is an alternative approach. If this approach works, we don't have to turn off `optimize-first-order-coercions?`.